### PR TITLE
Add separate target/scheme for ios pr build

### PR DESCRIPTION
**Pull request by Ferossgp**
We want to have a separate app identifier for PR build so we can use the release version on day to day basis, and be able to test new PRs without removing the release from the phone.

### CODE REVIEW

**Comment by andremedeiros** [comment]
@jakubgs can you jump in?

**Comment by jakubgs** [comment]
This is great, I tried to do that a year ago in https://github.com/status-im/status-react/pull/7479 but I could not fix the builds. Kept getting:
```
 ▸ ld: library not found for -lReact
```
Which drove me insane. I hope this works. Here's the PR: https://github.com/status-im/status-react-jenkins/pull/9
First use my branch name to the version of the `library` in all Jenkinsfiles:
https://github.com/status-im/status-react/blob/ddcb1fa54acd65ba755ac1e6ff8ddeb17c2993cb/ci/Jenkinsfile.ios#L1

**Comment by jakubgs** [comment]
I created two new provisioning profiles with `match`:

- https://github.com/status-im/ios-certificates/commit/cee66f0a
- https://github.com/status-im/ios-certificates/commit/a8575a0f

It should work now. I'll update the branch.

**Comment by jakubgs** [comment]
It doesn't seem to like the profile:
```
error: exportArchive: "StatusIm.app" requires a provisioning profile with the Associated Domains and Push Notifications features.
```
```
Error Domain=IDEProvisioningErrorDomain Code=9 
"StatusIm.app requires a provisioning profile with the Associated Domains and Push Notifications features."
UserInfo={
  IDEDistributionIssueSeverity=3,
  NSLocalizedDescription="StatusIm.app" requires a provisioning profile with the Associated Domains and Push Notifications features.,
  NSLocalizedRecoverySuggestion=Add a profile to the "provisioningProfiles" dictionary in your Export Options property list.
}
```
The profiles `match` creates don't have the right capapbilities:
ios_pr_profile

**Comment by jakubgs** [comment]
Looks like temporarily enabling automatic signing management and re-creating profiles fixed it:
Screenshot_2020-06-25_at_17 13 40

File: fastlane/Fastfile [hunk 1]
```diff
@@ -86,21 +86,34 @@ end
...
     clean: true,
     export_method: 'ad-hoc',
-    output_directory: 'status-adhoc'
+    output_name: 'StatusIm',
+    output_directory: 'status-adhoc',
+    export_options: {
+      signingStyle: 'manual',
+      provisioningProfiles: {
+        "im.status.ethereum": "match AdHoc im.status.ethereum",
+        "im.status.ethereum.pr": "match AdHoc im.status.ethereum.pr"
```
Comment (jakubgs) [inline fastlane/Fastfile:118]: Kinda annoying we need this, but ah well.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -86,21 +86,34 @@ end
 # `readonly`:
 #    if true  - only fetch existing certificates and profiles, don't upgrade from AppStoreConnect
 #    if false - read list of devices from AppStoreConnect, and upgrade the provisioning profiles from it
-def build_ios_adhoc(readonly)
+def build_ios_adhoc(readonly: false, pr_build: false)
+  # PR builds should appear as a separate App on iOS
+  scheme = pr_build ? 'StatusImPR' : 'StatusIm'
+  app_id = pr_build ? 'im.status.ethereum.pr' : 'im.status.ethereum'
```
Comment (jakubgs) [inline fastlane/Fastfile:96]: Can we maybe add a comment also explaining the `pr_build` argument the same way `readonly` is?

**Review by jakubgs — APPROVED** [review]
LGTM, tested on an iPhone too. Nice work!

File: ci/Jenkinsfile.macos [hunk 3]
```diff
@@ -1,4 +1,4 @@
-library 'status-react-jenkins@v1.1.7'
+library 'status-react-jenkins@ios-pr-builds'
```
Comment (jakubgs) [inline ci/Jenkinsfile.macos]: You didn't update this one, also there's other Jenkinsfiles in subfolders.

Comment (Ferossgp) [reply ci/Jenkinsfile.macos]: Should be correct now

**Review by jakubgs — APPROVED** [review]
you did it you crazy son of a bitch

**Pull request merged**